src/main/java/com/company/core/models/goods/Item.java

```diff
@@ -0,0 +1,104 @@
+package com.company.core.models.goods;
+
+import java.math.BigDecimal;
+import java.util.Objects;
+
+public class Item implements Identifiable {
+    private Long id;
+    private Order order;
+    private Product product;
+    private Integer quantity;
+    private BigDecimal price;
+
+    public Item(Product product, Integer quantity) {
+        this.price = BigDecimal.valueOf(0);
```

Comment:
Some classes that represent numbers have constants for the most commonly used values: it's better to use `BigDecimal.ZERO` here